### PR TITLE
[Asset] Add option $strictMode to JsonManifestVersionStrategy

### PR DESCRIPTION
**Pull request by GromNaN**
Documentation for symfony/symfony#38495

### CODE REVIEW

File: components/asset.rst [hunk 1]
```diff
@@ -160,11 +160,31 @@ In those cases, use the
...
+    $strictMode = true;
+    // assumes the JSON file above is called "rev-manifest.json"
+    $package = new Package(new JsonManifestVersionStrategy(__DIR__.'/rev-manifest.json'), $strictMode);
+
+    echo $package->getUrl('not-found.css');
+    // error:
+
+.. versionadded:: 5.2
+
+    Option ``$strictMode`` was introduced in Symfony 5.2.
```
Comment (OskarStark) [inline components/asset.rst]:
```suggestion
.. versionadded:: 5.3

    The ``$strictMode`` option was introduced in Symfony 5.3.
```

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -2243,11 +2251,23 @@ package:
...
+.. _reference-assets-strict-mode:
+
+strict_mode
+...........
+
+**type**: ``boolean`` **default**: ``false``
+
+.. versionadded:: 5.2
+
+    This option was introduced in Symfony 5.2.
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
.. versionadded:: 5.3

    This option was introduced in Symfony 5.3.
```

File: components/asset.rst [hunk 3]
```diff
@@ -160,11 +160,31 @@ In those cases, use the
...
+The ``$strictMode`` argument helps for debugging as it throws an exception when the asset is
+not listed in the manifest::
+
+    use Symfony\Component\Asset\Package;
+    use Symfony\Component\Asset\VersionStrategy\JsonManifestVersionStrategy;
+
+    // The value of $strict can be specific per environment "true" for debugging and "false" for stability.
+    $strictMode = true;
+    // assumes the JSON file above is called "rev-manifest.json"
+    $package = new Package(new JsonManifestVersionStrategy(__DIR__.'/rev-manifest.json'), $strictMode);
```
Comment (maxhelias) [inline components/asset.rst]: The $strictMode is on the VersionStrategy not on the Package

**Comment by javiereguiluz** [comment]
This been pending for too long. Apart from the rebase, is there anything else to be done to make this mergeable? Thanks!

**Comment by GromNaN** [comment]
I rebased an updated the doc according to the version it was merged.

**Review by maxhelias — APPROVED** [review]

**Pull request merged**